src/fastjet/_pyjet.py

```diff
@@ -228,3 +228,335 @@ def get_child(self, data):
         if not isinstance(data, ak.Array):
             raise TypeError("The input data is not an Awkward Array")
         return self._internalrep.get_child(data)
+
+
+class _FnDelayedInternalRepCaller:
+    def __init__(self, method_name, jetdef, **kwargs):
+        self.name = method_name
+        self.jetdef = jetdef
+        self.kwargs = kwargs
+
+    def __call__(self, array, *arrays):
+        if ak.backend(array) == "typetracer":
+            array.E.layout._touch_data(recursive=True)
+            array.px.layout._touch_data(recursive=True)
+            array.py.layout._touch_data(recursive=True)
+            array.pz.layout._touch_data(recursive=True)
+            for iarray in arrays:
+                iarray.E.layout._touch_data(recursive=True)
+                iarray.px.layout._touch_data(recursive=True)
+                iarray.py.layout._touch_data(recursive=True)
+                iarray.pz.layout._touch_data(recursive=True)
```

Comment:
I was about to say that this isn't sufficient because if the vector is made of a record with fields in a different coordinate system, `iarray.E`, etc. will be newly computed values that do not contain references to the data they were computed from. In that case,

```python
newly_computed_field._touch_data(recursive=True)
```

would do nothing.

_However,_ the act of computation _before_ you get the newly computed field touches the data, so this should be fine. It looks like `tests/test_008-dask.py::test_inclusive_from_file` does that test, and it passed.

So yeah—looks good to me!